### PR TITLE
fix(legacy) null check rrsets in domain

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Add extra null checks for the existence of `rrsets` when asserting if a domain can be deleted.

### MAAS deployment
To run this branch you will need access to one of the following MAAS deployments:
- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch
You can run this branch by:
- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps
- Go to /MAAS/l/domain/2
- Try and delete to the domain and see there are no errors in the console

## Fixes
Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/2132

## Backports
N/A

### CODE REVIEW

**Review by hatched — APPROVED** [review]

File: legacy/src/app/controllers/domain_details.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ function DomainDetailsController(
 
   // Called to check if the space can be deleted.
   $scope.canBeDeleted = function () {
-    if (angular.isObject($scope.domain)) {
+    if (angular.isObject($scope.domain) && $scope.domain.rrsets) {
```
Comment (huwshimi) [inline legacy/src/app/controllers/domain_details.js:132]: This could just be `if ($scope.domain?.rrsets) {`

Comment (huwshimi) [reply legacy/src/app/controllers/domain_details.js:132]: Also, is it possible this should remain unchanged and the line below should be: `return !$scope.domain.rrsets || $scope.domain.rrsets.length === 0;`?

**Comment by huwshimi** [comment]
Hey @anthonydillon just curious why you pushed this to upstream instead of your fork?

**Review by huwshimi — CHANGES_REQUESTED** [review]
@Caleb-Ellis and I have been unable to get the error on master so that we can confirm that this fixes it. Did you manage to reproduce it?

At any rate, I think `return !$scope.domain.rrsets || $scope.domain.rrsets.length === 0;` (or `return !$scope.domain?.rrsets?.length`) is probably a better fix as it looks like the intention here is to check that there are no `rrsets`.

I think it might also be sensible to add a test without the `rrsets` array: https://github.com/canonical-web-and-design/maas-ui/blob/master/legacy/src/app/controllers/tests/test_domain_details.js#L162

File: legacy/src/app/controllers/domain_details.js (hunk 1 above)
Comment (anthonydillon) [reply legacy/src/app/controllers/domain_details.js:132]: Sure, I would have used optional chaining in modern code but I didn't want to introduce a different way of coding in the middle of a file. This video is to be replaced in the near future. 

The second suggestion seems very complex to understand for the next developer that comes a long. I wanted to make it simple to read.

**Comment by anthonydillon** [comment]
I can add a test, just wasn't sure it was worth it for a null check in a code base that is being replaced

**Comment by huwshimi** [comment]
> I can add a test, just wasn't sure it was worth it for a null check in a code base that is being replaced

We tend to use the tests as a reference when rebuilding (or in the case of this utility probably porting it ). Also, in lieu of being able to QA the issue this would at least confirm that it fixes it.

**Review by huwshimi — APPROVED** [review]

**Comment by anthonydillon** [comment]
@huwshimi thanks, test added

**Comment by huwshimi** [comment]
Thanks @anthonydillon!
